### PR TITLE
Addition of truncate class to tile headers and when no deployments have started

### PR DESCRIPTION
**Pull request by sg00dwin**
Include ng-class to selectively apply margin-right for tiles that have rc shield & number
Tested in FF, Chrome, Safari, IE

Fixes https://github.com/openshift/origin-web-console/issues/1175

screen shot 2017-01-27 at 3 07 00 pm

### CODE REVIEW

**Comment by sg00dwin** [comment]
@rhamilto @spadgett

File: app/views/overview/_dc.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <div class="overview-tile" ng-class="{ 'deployment-in-progress': inProgressDeployment }">
   <ng-include src="'views/overview/_service-header.html'"></ng-include>
   <div class="overview-tile-header">
-    <div class="rc-header">
-      <div>
+    <div class="rc-header" ng-class="{ 'rc-header-shield' : activeReplicationController}">
```
Comment (spadgett) [inline app/views/overview/_dc.html]: You'll need to add the `rc-header-shield` class to the shield in _deployment.html as well.

**Comment by rhamilto** [comment]
Service names also have the same problem.  Want to fix that here?  Or another issue?

**Comment by sg00dwin** [comment]
@rhamilto @spadgett 
Added `rc-header-shield` to _deployment.html and service names are now truncated also.

Tested in FF, Chrome, Safari, IE

screen shot 2017-01-30 at 2 18 01 pm

File: app/views/overview/_deployment.html [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <div class="overview-tile" ng-class="{ 'deployment-in-progress': inProgressDeployment }">
   <ng-include src="'views/overview/_service-header.html'"></ng-include>
   <div class="overview-tile-header">
-    <div class="rc-header">
-      <div>
+    <div class="rc-header" ng-class="{ 'rc-header-shield' : activeReplicationController}">
```
Comment (spadgett) [inline app/views/overview/_deployment.html]:
```html
<div class="rc-header" ng-class="{ 'rc-header-shield': latestReplicaSet && latestRevision && !inProgressDeployment }">
```

Comment (spadgett) [inline app/views/overview/_deployment.html]: @sg00dwin You can test with

https://github.com/openshift/origin-web-console/blob/master/test/testdata/k8s-deployments.yaml

**Comment by spadgett** [comment]
@sg00dwin I see you pushed some updates. Ready for another review?

**Comment by sg00dwin** [comment]
@spadgett Yes please review.

I enabled the truncation of service names. To get that to work, required some changes to `overview-service` and `.service-name` and removing the flex from tiles for `.no-deployments-message` and  then applied the `overview-tile` rules, so that they wouldn't over expand. Also, the truncation along with  `line-height: 1.1` inherited from `.h3()` caused clipping of text descenders. So I increased line-height and offset the additional height by reducing the top and bottom margins so it maintained it's current height.

Additionally
Since the links in `.empty-tile` and `.empty-dc` where in the middle of a sentence I couldn't truncate so I applied `.word-break-all` so that they would wrap in every browser. Since this can cause weird breaking mid word, I reduced the left and right padding so that it would allow more room for long strings and reduce the chance of it occurring.


screen shot 2017-01-31 at 4 55 06 pm
screen shot 2017-01-31 at 4 52 45 pm

File: app/views/overview/_service.html [hunk 3]
```diff
@@ -1,11 +1,10 @@
 <div ng-if="!tileCount" class="no-deployments-block">
-  <div column class="no-deployments-message">
+  <div column class="no-deployments-message overview-tile">
     <ng-include src="'views/overview/_service-header.html'"></ng-include>
     <div class="empty-tile">
       <h2>No deployments or pods.</h2>
       <p>
-        Service
-        <a ng-href="{{service | navigateResourceURL}}">{{service.metadata.name}}</a>
+        <div class="word-break-all">Service <a ng-href="{{service | navigateResourceURL}}">{{service.metadata.name}}</a></div>
```
Comment (spadgett) [inline app/views/overview/_service.html]: Isn't this going to add a line break in the middle of a sentence?

File: app/views/overview/_service-group.html [hunk 4]
```diff
@@ -71,12 +71,10 @@ <h3 class="route-title truncate">
             <div class="no-child-services-message">
               <div class="empty-tile">
                 <h2>No grouped services.</h2>
-                <p>
-                  No services are grouped with <a ng-href="{{service | navigateResourceURL}}">{{service.metadata.name}}</a>.
+                  No services are grouped with <div class="word-break-all"><a ng-href="{{service | navigateResourceURL}}">{{service.metadata.name}}</a>.</div>
```
Comment (spadgett) [inline app/views/overview/_service-group.html]: Isn't this going to add a line break in the middle of a sentence?

File: app/views/overview/_dc.html [hunk 5]
```diff
@@ -43,8 +43,8 @@
         A new deployment will start automatically when
         <span ng-if="imageChangeTriggers.length === 1">
           an image is available for
-          <a ng-href="{{urlForImageChangeTrigger(imageChangeTriggers[0], deploymentConfig)}}">
-            {{imageChangeTriggers[0].imageChangeParams.from | imageObjectRef : deploymentConfig.metadata.namespace}}</a>.
+          <div class="truncate mar-bottom-xl"><a ng-href="{{urlForImageChangeTrigger(imageChangeTriggers[0], deploymentConfig)}}">
```
Comment (spadgett) [inline app/views/overview/_dc.html]: Isn't this going to add a line break in the middle of a sentence?

File: app/views/overview/_service.html (hunk 3 above)
Comment (sg00dwin) [inline app/views/overview/_service.html]: yep, these will be fixed.

**Comment by sg00dwin** [comment]
@spadgett PTAL

**Comment by sg00dwin** [comment]
screen shot 2017-02-02 at 3 36 29 pm

**Comment by spadgett** [comment]
This change breaks sentences mid-word in Firefox. Can we use `word-break` instead of `word-break-all`?

openshift_web_console

**Comment by sg00dwin** [comment]
With just `<p class="word-break">...` There would be scenarios where the long string wouldn't break and cause the tile width to expand, causing it to bump down below it's sibling tile.

screen shot 2017-02-06 at 10 43 22 am

**Comment by sg00dwin** [comment]
I'm looking further to see if we can incorporate [hyphens](https://developer.mozilla.org/en-US/docs/Web/CSS/hyphens) with our `word-break` rule

**Comment by sg00dwin** [comment]
@spadgett  I have `.break-word` working  in those empty tiles, ftw!
Cross browser screenshots...

screen shot 2017-02-06 at 3 52 21 pm
screen shot 2017-02-06 at 3 51 43 pm
screen shot 2017-02-06 at 3 46 39 pm
screen shot 2017-02-06 at 3 44 57 pm

**Comment by rhamilto** [comment]
#1205 is a related bug that can also be fixed in this PR.

**Comment by spadgett** [comment]
@sg00dwin Approved, but needs rebase

**Comment by sg00dwin** [comment]
@spadgett this is now rebased

**Comment by sg00dwin** [comment]
@spadgett removed vendor.js

**Comment by spadgett** [comment]
[merge]

**Pull request merged**